AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/Model+Compare.swift

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+// swiftlint:disable cyclomatic_complexity
+extension ModelSchema {
+
+    private static let serviceUpdatedFields: Set = ["updatedAt"]
```

Comment:
- can use a `Set` for faster look up
- is it possible to add `createdAt` as an explicit and optional field? resulting in `createdAt` not a read-only and optionally passed in when creating the model. if this case is true, then should add `createdAt` into the Set as well

Reply:
it seems logical to add `createdAt` to the list - will do.
is there a faster `Set` implementation?